### ISSUE TITLE
revokeAccessToken causes a NPE with the following stack trace

### ISSUE THREAD

**Issue by manrock007** [thatJavaNerd/JRAW#155]
I'm trying to integrate the logout functionality(Hope I'm using the right API for log out..) It causes a NullPointerException in the HttpLogger class..

```

08-14 20:36:58.808 29767-31715/com.dreamcast.redditr E/ACRA: ACRA caught a RuntimeException for com.dreamcast.redditr
                                                             java.lang.RuntimeException: An error occurred while executing doInBackground()
                                                                 at android.os.AsyncTask$3.done(AsyncTask.java:309)
                                                                 at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
                                                                 at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
                                                                 at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                                                                 at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
                                                                 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                                                                 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                                                                 at java.lang.Thread.run(Thread.java:818)
                                                              Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.String.replace(java.lang.CharSequence, java.lang.CharSequence)' on a null object reference
                                                                 at net.dean.jraw.http.HttpLogger.log(HttpLogger.java:261)
                                                                 at net.dean.jraw.http.RestClient.execute(RestClient.java:123)
                                                                 at net.dean.jraw.RedditClient.execute(RedditClient.java:143)
                                                                 at net.dean.jraw.RedditClient.execute(RedditClient.java:137)
                                                                 at com.dreamcast.redditr.jraw.AndroidRedditClient.execute(AndroidRedditClient.java:82)
                                                                 at net.dean.jraw.http.oauth.OAuthHelper.revokeToken(OAuthHelper.java:285)
                                                                 at net.dean.jraw.http.oauth.OAuthHelper.revokeAccessToken(OAuthHelper.java:261)
                                                                 at com.dreamcast.redditr.models.RAccountManager$4.doInBackground(RAccountManager.java:207)
                                                                 at com.dreamcast.redditr.models.RAccountManager$4.doInBackground(RAccountManager.java:203)
                                                                 at android.os.AsyncTask$2.call(AsyncTask.java:295)
                                                                 at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                 at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234) 
                                                                 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
                                                                 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
                                                                 at java.lang.Thread.run(Thread.java:818) 
```

**Comment by thatJavaNerd** [comment]
Can you post the relevant code?

**Comment by manrock007** [comment]
Code where I invoke the revokeAccessToken method. The reddit instance is the same singleton instance that I use across the app.. 
```
// RAccountManager.java
    public void logout() {
        new AsyncTask<Credentials,Void,Void>() {

            @Override
            protected Void doInBackground(Credentials... params) {
                App.getInstance().getReddit().getOAuthHelper().revokeAccessToken(App.CREDENTIALS);
                return null;
            }

            @Override
            protected void onPostExecute(Void aVoid) {
                super.onPostExecute(aVoid);
                sharedpreferences.edit().clear().commit();
            }
        }.execute();
    }
```

```
// App.java
public static final Credentials CREDENTIALS = Credentials.installedApp("XXXXXX-XXXXXXXX", "http://localhost:4515/signin-reddit");
```

**Comment by thatJavaNerd** [comment]
It seemed HttpLogger didn't check to see if the raw response was null or not before operating on it. Thanks for pointing this out!

**Comment by amzgolinski** [comment]
has this fix been published yet?  I am using 0.9.0 (with android client) and am still seeing the exception

**Comment by karanrajpal14** [comment]
@manrock007 any luck with implementing logout? I'm trying to do the same and I can't get it to work. Facing the same issue as you.
